examples/add-role.py

```diff
@@ -0,0 +1,198 @@
+    """
+    Return the ID of the user with the given email.
+
+    :param email: Email address of the user.
+    :return: Matching user's ID. None if the user cannot be found.
+    """
+    response = api.search_users(search_payload={"keywords": EMAIL})
+
+    if not response.ok:
+        print(response.data)
+        sys.exit(1)
+
+    for item in response.data.get("items"):
+        if item.get("email") == EMAIL:
+            return item.get("id")
+    else:
+        return None
+
+
+def get_user_roles(user_id: str) -> list:
+    """
+    Return the current roles of the target user.
+
+    :param user_id: ID of the target user.
+    :return: List of target-user's roles.
+    """
+    response = api.get_user_roles(user_id)
+
+    if not response.ok:
+        print(response.data)
+        sys.exit(1)
+
+    return response.data.get("items")
+
+
+def get_role_ID(name: str) -> str:
+    """
+    Return the ID of the named role.
+
+    :param name: Name of the role.
+    :return: Matching role's ID. None if the role cannot be found.
+    """
+    response = api.get_roles()
+
+    if not response.ok:
+        print(response.data)
+        sys.exit(1)
+
+    for role in response.data.get("items"):
+        if role.get("name") == ROLE:
+            return role.get("id")
+    else:
+        return None
+
+
+def get_new_role(role_id: str, grant_type: str, duration=1) -> object:
+    """
+    Return the role object for granting target-role membership.
+
+    :param role_id: ID of the target role.
+    :param grant_type: Grant type for specifying membership validity.
+    :param duration: Validity period of the grant.
+                     Only applicable to non-permanent grant types.
+
+    :return: Role object for the specified grant.
+    """
+    start = datetime.datetime.utcnow()
+    end = start + datetime.timedelta(hours=duration)
+
+    roles = {
+        "PERMANENT":
+        {
+            "id": role_id,
+            "name": ROLE,
+            "grant_type": "PERMANENT",
+            "grant_start": None,
+            "grant_end": None,
+            "floating_length": None,
+            "explicit": True
+        },
+        "FLOATING":
+        {
+            "id": role_id,
+            "name": ROLE,
+            "grant_type": "FLOATING",
+            "grant_start": None,
+            "grant_end": None,
+            "floating_length": duration,
+            "explicit": True
+        },
+        "TIME_RESTRICTED":
+        {
+            "id": role_id,
+            "name": ROLE,
+            "grant_type": "TIME_RESTRICTED",
+            "grant_start": start.isoformat() + "Z",
+            "grant_end": end.isoformat() + "Z",
+            "floating_length": 0,
+            "explicit": True
+        }
+    }
+
+    if grant_type not in roles:
+        print(f"Invalid grant type {grant_type}. Exiting...")
+        sys.exit(1)
+
+    return roles[grant_type]
+
+
+def main():
+    print(f"Searching for user {EMAIL}")
+    userID = get_user_ID(EMAIL)
```

Comment:
We are using a style that calls for all lowercase for variable and function names. Also function name should be all lowercase except for acronyms. Eg: user_id, get_user_id, etc...

Reply:
Lowercased functions.

Did not lowercase the global variables: Looking at other examples we use uppercase for any variables that the user is expected to set for their own needs.